include/RAJA/policy/openmp/reduce.hpp

```diff
@@ -69,497 +73,160 @@
+      Reduce()(parent->my_data, my_data);
     }
   }
 
-  //! return the reduced min value.
-  /*!
-   *  \return the calculated reduced value
-   */
-  RAJA_HOST_DEVICE operator T() {
-    return m_val;
-  }
+  void combine(T const &other) { Reduce{}(my_data, other); }
+  void combine(T const &other) const { Reduce{}(my_data, other); }
 
-  //! return the reduced min value.
   /*!
    *  \return the calculated reduced value
    */
-  RAJA_HOST_DEVICE T get() {
-    return operator T();
-  }
-
-  //! reducer function; updates the current instance's state
-  /*!
-   * Assumes each thread has its own copy of the object.
-   */
-  RAJA_HOST_DEVICE const ReduceMin &min(T rhs) const
-  {
-    Reduce()(m_val, rhs);
-    return *this;
-  }
+  operator T() const { return my_data; }
 
-  //! reducer function; updates the current instance's state
   /*!
-   * Assumes each thread has its own copy of the object.
+   *  \return the calculated reduced value
    */
-  RAJA_HOST_DEVICE ReduceMin &min(T rhs)
-  {
-    Reduce()(m_val, rhs);
-    return *this;
-  }
-
-private:
-  //! pointer to the parent ReduceMin object
-  const ReduceMin *m_parent;
-  mutable T m_val;
+  T get() const { return operator T(); }
 };
 
-/*!
- **************************************************************************
- *
- * \brief  MinLoc reducer class template for use in OpenMP execution.
- *
- **************************************************************************
- */
-template <typename T>
-class ReduceMinLoc<omp_reduce, T>
+template <typename T, typename Reduce>
+class ReduceOMPOrdered
 {
-  using Reduce = RAJA::reduce::minloc<T, Index_type>;
+  ReduceOMPOrdered *parent = nullptr;
+  std::shared_ptr<std::vector<T>> data;
+  T identity;
+  T mutable my_data;
 
 public:
+  using value_type = T;
   //! prohibit compiler-generated default ctor
-  ReduceMinLoc() = delete;
+  ReduceOMPOrdered() = delete;
 
-  //! prohibit compiler-generated copy assignment
-  ReduceMinLoc &operator=(const ReduceMinLoc &) = delete;
-
-  //! compiler-generated move constructor
-  ReduceMinLoc(ReduceMinLoc &&) = default;
-
-  //! compiler-generated move assignment
-  ReduceMinLoc &operator=(ReduceMinLoc &&) = default;
-
-  //! constructor requires a default value for the reducer
-  RAJA_HOST_DEVICE explicit ReduceMinLoc(T init_val, Index_type init_idx)
-      : m_parent(nullptr), m_val(init_val), m_idx(init_idx)
+  constexpr ReduceOMPOrdered(T init_val, T identity_ = T())
+      : parent{this},
+        data{
+            std::make_shared<std::vector<T>>(omp_get_max_threads(), identity_)},
+        identity{identity_},
+        my_data{init_val}
   {
   }
-
-  //! create a copy of the reducer
-  /*!
-   * keep parent the same if non-null or set to current
-   */
-  RAJA_HOST_DEVICE ReduceMinLoc(const ReduceMinLoc &other)
-      : m_parent(other.m_parent ? other.m_parent : &other),
-        m_val(other.m_val),
-        m_idx(other.m_idx)
+  constexpr ReduceOMPOrdered(ReduceOMPOrdered const &other)
```

Comment:
I'm going to stop with the 'blank line b4' comments to avoid being increasingly annoying....